float32 jointA
float32 jointB
float32 jointC
float32 jointD